README.md

```diff
@@ -89,455 +81,113 @@ opam switch list-available
-is not working well with multiple domains.
-
-```ocaml
-type queue = int list ref
-
-let create () : queue = ref []
-
-let push q a = q := a :: !q
-```
-
-What happens if we try to use this queue with multiple domains? First, let's
-define the work we want a single domain to do: each domain will push 10 times
-its `id` in the queue.
-
-```ocaml
-let work id q = for i = 0 to 9 do push q id done
-```
-
-Then let's define our test : it spawns 2 domains that each execute `work` in
-parallel. `test` returns the content of the queue as well as its length, so we
-can easily see if it contains the 20 elements we expect.
-
-```ocaml
- let test () =
-   let q = create () in
-   let domainA = Domain.spawn (fun () -> work 1 q) in
-   let domainB = Domain.spawn (fun () -> work 2 q) in
-   Domain.join domainA;
-   Domain.join domainB;
-   (List.length !q, !q)
-```
-
-Let's try it :
-
-```ocaml
-# test ()
-- : int * int list =
-(20, [2; 2; 2; 2; 2; 2; 2; 2; 2; 2; 1; 1; 1; 1; 1; 1; 1; 1; 1; 1])
-```
-
-Everything seems fine, right? Except, it is not running in parallel, as we can
-see from the consecutive `2` (pushed by `domainB`) and `1` pushed by `domainA`.
-This is because spawning a domain takes way more time than executing `work`, so
-`domainA` is long finished before `domainB` is even spawned. One way to overpass
-this issue is to increase the amount of work done in `work` (for example, by
-pushing more elements). Another way is to make sure the domains wait for each
-other before beginning their workload.
+# Provided data structures
 
-We use a basic
-[barrier implementation](https://github.com/ocaml-multicore/saturn/blob/main/test/barrier/barrier.mli)
-to do that. Each domain will now wait for the other to reach the barrier before
-beginning to push.
+### Treiber Lock-free Stack
+- **Module**: [Stack](https://ocaml-multicore.github.io/saturn/saturn/Stack/index.html)
+- **Description**: A classic multi-producer, multi-consumer, lock-free stack, known for robustness and flexibility.
+- **Recommendation**: It's a recommended starting point when a LIFO structure is needed.
 
-```ocaml
-let work_par id barrier q =
-  Barrier.await barrier;
-  for i = 0 to 9 do
-    push q id
-  done
-```
-
-The `test` function is now:
+### Lock-free Bounded Stack
 
-```ocaml
-let test_par () =
-  let barrier = Barrier.create 2 in
-  let q = create () in
-  let domainA = Domain.spawn (fun () -> work_par 1 barrier q) in
-  let domainB = Domain.spawn (fun () -> work_par 2 barrier q) in
-  Domain.join domainA;
-  Domain.join domainB;
-  (List.length !q, !q)
-```
+- **Module**: [Bounded_stack](https://ocaml-multicore.github.io/saturn/saturn/Bounded_stack/index.html)
+- **Description**: A stack based on the Treiber stack algorithm, with a limited capacity and a `length` function.
+- **Recommendation**: Adding a capacity introduces a general overhead to the operations. It is recommended to use the unbounded stack if neither the capacity nor the `length` function is needed.
```

Comment:
Perhaps explain the positive side of having a max capacity?

Reply:
It's memory bounded. Are you thinking of anything else ?

Reply:
No that's it!